### PR TITLE
Censoring Internal Java Solution

### PR DESCRIPTION
**Pull request by brad182**

### CODE REVIEW

File: solutions/usaco-526.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+id: usaco-526
+source: USACO Bronze 2015 February
+title: Loan Repayment
```
Comment (maggie-j-liu) [inline solutions/usaco-526.mdx:4]: wrong title?

File: solutions/usaco-526.mdx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_censor_bronze.html)
+
+## Implementation
+
+**Time Complexity:** 
+$\mathcal{O}(ST)$
```
Comment (maggie-j-liu) [inline solutions/usaco-526.mdx:13]:
```suggestion
**Time Complexity:** $\mathcal{O}(ST)$
```

File: solutions/usaco-526.mdx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		Kattio io = new Kattio("censor");
+		String string = io.next();
+		String toCensor = io.next();
+
+		// use StringBuilder in Java because it is significantly faster
+		StringBuilder s = new StringBuilder();
+		s.append(string.substring(0, toCensor.length() - 1));
+
+		// try adding one character at a time
+		for (int x = (toCensor.length() - 1); x < string.length(); x++) {
```
Comment (maggie-j-liu) [inline solutions/usaco-526.mdx:34]:
```suggestion
		for (int x = toCensor.length() - 1; x < string.length(); x++) {
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-526.mdx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Censoring {
```
Comment (SansPapyrus683) [inline solutions/usaco-526.mdx:23]:
```suggestion
public class Censor {
```

File: solutions/usaco-526.mdx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+			// need to check if our current string is longer than the censored word
+			if (s.length() >= toCensor.length()) {
+				String check = s.substring(s.length() - toCensor.length());
+				// delete the censored word if present
+				if (check.equals(toCensor)) {
+					s.delete(s.length() - toCensor.length(), s.length());
+				}
+			}
+		}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-526.mdx:45]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-526.mdx (hunk 1 above)
Comment (brad182) [reply solutions/usaco-526.mdx:4]: oops i copy paste from my old prs

**Review by SansPapyrus683 — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**